mods/RFA/hook/lua/ui/controls/worldview.lua

```diff
@@ -238,26 +356,44 @@ do
             TableClear(self._selectionRings)
         end,
 
-        UpdateBuildRings = function(self)
+        ---@param self WorldView
+        ---@param useMousePos boolean
+        UpdateBuildRings = function(self, useMousePos)
             local selection = GetSelectedUnits()
             if not selection then
                 self:ClearBuildRings()
                 return
             end
-            local engineers = EntityCategoryFilterDown(categories.ENGINEER, selection)
-            if TableEmpty(engineers) then
+            ---@type UserUnit[]
+            local builders = EntityCategoryFilterDown(buildersCategory, selection)
+            if TableEmpty(builders) then
                 self:ClearBuildRings()
                 return
             end
 
             local radius = selection
```

Comment:
If you use `builders` in picking position, should you pick radius for `builders` too?

Reply:
The result is the same but the logic is somewhat different imo: you shouldn't check build radius for units that can't build